extension/mozcentral/browser/extensions/shumway/content/ShumwayUtils.jsm

```diff
@@ -56,6 +58,19 @@ let ShumwayUtils = {
     Services.prefs.addObserver(PREF_DISABLED, this, false);
   },
 
+  migratePreferences: function migratePreferences() {
+    // At one point we had shumway.disabled set to true by default,
+    // and we are trying to replace it with shumway.swf.whitelist:
+    // checking if the user already changed it before to reset
+    // the whitelist to '*'.
+    if (Services.prefs.prefHasUserValue(PREF_DISABLED) &&
+        !Services.prefs.prefHasUserValue(PREF_WHITELIST) &&
+        !getBoolPref(PREF_DISABLED, false)) {
+      // The user is already using Shumway -- enabling all web sites.
+      Services.prefs.setCharPref(PREF_WHITELIST, '*');
```

Comment:
For posterity's sake: we discussed this on IRC and I agree that, while it sucks, this is the most sensible way to deal with the issue.